pkg/cloud/azure/services/internalloadbalancers/internalloadbalancers.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internalloadbalancers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-12-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/pkg/errors"
+	"k8s.io/klog"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services/subnets"
+)
+
+// Spec specification for internal load balancer
+type Spec struct {
+	Name       string
+	SubnetName string
+	VNETName   string
+	IPAddress  string
+}
+
+// Get provides information about a route table.
+func (s *Service) Get(ctx context.Context, spec interface{}) (interface{}, error) {
```

Comment:
Why isn't spec a concrete type?

Reply:
Moved to an interface type for now, may be later it can be extended in future